### PR TITLE
Fix: Tokenizer splits abbreviations with periods into individual characters (#216)

### LINKED ISSUE #216: Tokenizer splits abbreviations with periods into individual characters
## Description

The tokenizer incorrectly splits abbreviations containing periods (like "c.e.o.s" and "U.S.") into individual character tokens instead of treating them as single words.

## Current Behavior

```python
from services.tokenizer import tokenize_text

# Abbreviations are incorrectly split
text = "The c.e.o.s met yesterday"
result = tokenize_text(text)
print(result)
# Output: ["the", "c", "e", "o", "s", "met", "yesterday"]

# Same issue with country codes
text = "I live in U.S. now"
result = tokenize_text(text)
print(result)
# Output: ["i", "live", "in", "u", "s", "now"]
```

## Expected Behavior

```python
# Abbreviations should be kept as single tokens
text = "The c.e.o.s met yesterday"
result = tokenize_text(text)
print(result)
# Expected: ["the", "c.e.o.s", "met", "yesterday"]

text = "I live in U.S. now"
result = tokenize_text(text)
print(result)
# Expected: ["i", "live", "in", "u.s.", "now"]
```

## Root Cause

**Location**: `services/tokenizer/basic/patterns.py:101`

The `LATIN_WORD_PATTERN` regex only matches letters and apostrophes (for contractions):

```python
LATIN_WORD_PATTERN = r"[a-zA-Z]+(?:\'[a-zA-Z]+)*"  # Handle contractions
```

This pattern does **not include periods**, so abbreviations like "c.e.o.s" are tokenized as:
- "c" (word) + "." (punctuation) + "e" (word) + "." (punctuation) + ...

Since the default configuration has `include_punctuation=False`, the periods are filtered out in post-processing, leaving only individual letters.

## Impact

This bug affects:
- **Abbreviations**: c.e.o.s, U.S., U.K., Ph.D., etc.
- **Acronyms**: I.B.M., N.A.S.A., etc.
- **Initialisms**: Any text with period-separated letters

**Not affected** (working correctly):
- Chinese character tokenization (uses different pattern)
- Contractions like "don't", "it's" (apostrophe support already exists)
- Regular words without punctuation

## Proposed Solution

Update `LATIN_WORD_PATTERN` in `services/tokenizer/basic/patterns.py:101`:

```python
# Current:
LATIN_WORD_PATTERN = r"[a-zA-Z]+(?:\'[a-zA-Z]+)*"

# Proposed:
LATIN_WORD_PATTERN = r"[a-zA-Z]+(?:\.[a-zA-Z]+)*\.?(?:\'[a-zA-Z]+)*"
```

**Pattern breakdown:**
- `[a-zA-Z]+` - One or more letters (required)
- `(?:\.[a-zA-Z]+)*` - Zero or more: period + letters (handles "c.e.o.s", "U.S.A")
- `\.?` - Optional trailing period (handles "U.S.")
- `(?:\'[a-zA-Z]+)*` - Existing contraction support (handles "don't")

**Why this works:**
- Matches abbreviations as single tokens: "c.e.o.s" → `["c.e.o.s"]`
- Matches country codes: "U.S." → `["u.s."]`
- Preserves existing contraction support: "don't" → `["don't"]`
- Ellipses ("...") handled separately by `PUNCTUATION_PATTERN`
- No impact on CJK character-level tokenization

## Test Cases Required

```python
class TestAbbreviationsAndPunctuation:
    """Test abbreviation handling and punctuation edge cases."""

    def test_abbreviations_basic(self):
        """Basic abbreviation: c.e.o.s"""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("The c.e.o.s met yesterday")
        assert result == ["the", "c.e.o.s", "met", "yesterday"]

    def test_abbreviations_with_trailing_period(self):
        """Abbreviation with trailing period: U.S."""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("I live in U.S. now")
        assert result == ["i", "live", "in", "u.s.", "now"]

    def test_multiple_abbreviations(self):
        """Multiple abbreviations in one sentence"""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("U.S. and U.K. relations")
        assert result == ["u.s.", "and", "u.k.", "relations"]

    def test_ellipses_without_punctuation(self):
        """Ellipses should be filtered when punctuation disabled"""
        tokenizer = BasicTokenizer()  # Default: include_punctuation=False
        result = tokenizer.tokenize("Wait for it...")
        assert result == ["wait", "for", "it"]

    def test_chinese_tokenization_regression(self):
        """Regression: Chinese character-level tokenization"""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("你好世界")
        assert result == ["你", "好", "世", "界"]

    def test_contractions_regression(self):
        """Regression: Contractions still work"""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("I don't think it's ready")
        assert result == ["i", "don't", "think", "it's", "ready"]

    def test_abbreviations_and_contractions_together(self):
        """Abbreviations and contractions in same text"""
        tokenizer = BasicTokenizer()
        result = tokenizer.tokenize("U.S. citizens don't always agree")
        assert result == ["u.s.", "citizens", "don't", "always", "agree"]
```

## Checklist

- [ ] Update `LATIN_WORD_PATTERN` in `patterns.py`
- [ ] Add comprehensive test cases in `test_basic_tokenizer.py`
- [ ] Run full test suite to verify no regressions
- [ ] Verify Chinese tokenization still works (character-level)
- [ ] Verify contraction handling still works (apostrophes)
- [ ] Verify ellipses handling with/without `include_punctuation`

## Files to Modify

1. `services/tokenizer/basic/patterns.py` (line 101)
2. `services/tokenizer/basic/test_basic_tokenizer.py` (add new test class)

### PR DESCRIPTION
**Pull request by JoeKarow**
## Summary

Fixes tokenizer bug where abbreviations containing periods (like "c.e.o.s" and "U.S.") are incorrectly split into individual character tokens instead of being treated as single words.

### 1. Updated `LATIN_WORD_PATTERN` regex (`patterns.py:101`)
- **Before:** `r"[a-zA-Z]+(?:\'[a-zA-Z]+)*"`
- **After:** `r"[a-zA-Z]+(?:\.[a-zA-Z]+)+\.?|[a-zA-Z]+(?:\'[a-zA-Z]+)*"`

The new pattern supports both:
- **Abbreviations**: "U.S.", "c.e.o.s", "Ph.D." (period-separated letters)
- **Contractions**: "don't", "it's" (apostrophe-separated words)

### 2. Fixed `_is_url_like()` method (`tokenizer.py:216-245`)
Added logic to distinguish abbreviations from URLs:
- Checks for explicit URL markers (http://, https://, www., ://)
- Uses abbreviation pattern `^[a-z]{1,3}(?:\.[a-z]{1,3})+\.?$` to prevent abbreviations from being classified as URLs
- Prevents trailing periods from being stripped from abbreviations like "U.S."

### 3. Added comprehensive test suite (TDD approach)
New test class `TestAbbreviationsAndPunctuation` with 7 tests:
- ✅ Basic abbreviations ("c.e.o.s")
- ✅ Abbreviations with trailing periods ("U.S.")
- ✅ Multiple abbreviations in one sentence
- ✅ Ellipses handling (regression check)
- ✅ Chinese character tokenization (regression check)
- ✅ Contractions (regression check)
- ✅ Mixed abbreviations and contractions

## Test Results

**All tests passing:**
- 7/7 new abbreviation tests ✅
- 105/105 total tokenizer tests ✅
- No regressions in existing functionality ✅

**Examples now working correctly:**
```python
"c.e.o.s met"  → ["c.e.o.s", "met"]           ✅
"U.S. and U.K." → ["u.s.", "and", "u.k."]    ✅
"Ph.D. program" → ["ph.d.", "program"]        ✅
"don't"         → ["don't"]                    ✅ (still works)
"你好世界"      → ["你", "好", "世", "界"]    ✅ (still works)
```

## Impact

- **Fixes:** Abbreviation tokenization for period-separated letter sequences
- **Preserves:** All existing functionality (contractions, CJK/Arabic/Thai tokenization, social media entities)
- **No breaking changes:** Pattern extends functionality without changing existing behavior

## Closes

Closes #216

### CODE REVIEW

File: services/tokenizer/basic/tokenizer.py [hunk 1]
```diff
@@ -219,15 +219,32 @@ def _is_url_like(self, token: str) -> bool:
...
+        # Heuristic: URLs have at least one period NOT followed by a single uppercase/lowercase letter
+        # This allows "example.com" but excludes "U.S." and "c.e.o.s"
+        if (
+            token.count(".") >= 1
+            and any(c.isalpha() for c in token)
+            and "@" not in token
+        ):
+            # Check if this looks like an abbreviation (single letters between periods)
+            # Pattern: letter(s).letter(s).letter(s) where segments are 1-3 chars
+            # Check if this looks like an abbreviation (single letters between periods)
```
Comment (KristijanArmeni) [inline services/tokenizer/basic/tokenizer.py:237]: This comment seems redundant (seems like duplication of the line above)?

**Review by KristijanArmeni — CHANGES_REQUESTED** [review]
This looks good to go @JoeKarow, thank you! There's just some small cleanup needed with duplicated comments/code before we merge this in.

File: services/tokenizer/basic/tokenizer.py (hunk 1 above)
Comment (JoeKarow) [reply services/tokenizer/basic/tokenizer.py:235]: Yeah - it got duplicated in error
